cmd/svcat/completion/completion_cmd.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package completion
+
+import (
+	"fmt"
+	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/command"
+	"github.com/spf13/cobra"
+)
+
+const defaultBoilerPlate = `
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+`
+
+var (
+	completionLong = `
+Output shell completion code for the specified shell (bash).
+The shell code must be evaluated to provide interactive
+completion of svcat commands. This can be done by sourcing it from
+the .bash_profile.
+
+Note: this requires the bash-completion framework, which is not installed
+by default on Mac. This can be installed by using homebrew:
+
+	$ brew install bash-completion
+
+Once installed, bash_completion must be evaluated. This can be done by adding the
+following line to the .bash_profile
+
+	$ source $(brew --prefix)/etc/bash_completion
+`
+
+	completionExample = `
+# Install bash completion on a Mac using homebrew
+brew install bash-completion
+printf "\n# Bash completion support\nsource $(brew --prefix)/etc/bash_completion\n" >> $HOME/.bash_profile
+source $HOME/.bash_profile
+
+# Load the svcat completion code for bash into the current shell
+source <(svcat completion bash)
+
+# Write bash completion code to a file and source if from .bash_profile
+svcat completion bash > ~/.svcat/svcat_completion.bash.inc
+printf "\n# Svcat shell completion\nsource '$HOME/.svcat/svcat_completion.bash.inc'\n" >> $HOME/.bash_profile
+source $HOME/.bash_profile
+`
+)
+
+var (
+	completionShells = map[string]func(cxt *command.Context, cmd *cobra.Command) error{
+		"bash": runCompletionBash,
+	}
+)
+
+// NewCompletionCmd return command for executing "svcat completion" command
+func NewCompletionCmd(cxt *command.Context, boilerPlate string) *cobra.Command {
```

Comment:
I recommend removing the `boilerPlate` parameter and then just not include a header in the output. It's generated by cobra anyway, so I don't feel like it needs a copyright on it anyway.